### PR TITLE
Payment method as array

### PR DESCRIPTION
**Pull request by chris-redbeed**
I replace the comma string with an array.
The comma string don't work and the Class only allowed stings.

With this changes it is possible to set up multiple method correctly.

### CODE REVIEW

**Comment by sandervanhooft** [comment]
A lot of people will be using the current default config for this, so we'll need to provide something that's backwards compatible.

When processing the method from the config, how about:

If null: convert to empty array []
If string: remove spaces, explode(',')

Then process the array.

File: src/FirstPayment/FirstPaymentBuilder.php [hunk 1]
```diff
@@ -6,10 +6,14 @@
 use Illuminate\Support\Str;
 use Laravel\Cashier\Cashier;
 use Laravel\Cashier\FirstPayment\Actions\ActionCollection;
+use Laravel\Cashier\Traits\PaymentMethodString;
```
Comment (sandervanhooft) [inline src/FirstPayment/FirstPaymentBuilder.php:9]: This trait is specific to the FirstPayment, so please move it to the `Laravel\Cashier\FirstPayment` namespace.

File: src/Plan/Plan.php [hunk 2]
```diff
@@ -6,10 +6,14 @@
 
 use Laravel\Cashier\Order\OrderItemPreprocessorCollection;
 use Laravel\Cashier\Plan\Contracts\Plan as PlanContract;
+use Laravel\Cashier\Traits\PaymentMethodString;
```
Comment (sandervanhooft) [inline src/Plan/Plan.php:9]: This trait is specific to the FirstPayment, so please move it to the Laravel\Cashier\FirstPayment namespace.

**Comment by chris-redbeed** [comment]
@sandervanhooft FYI: Currently I didn't run the unit tests because I need to set up a test env. with the mandatory IDs (Payment IDs, Customer IDs) for the test. I will let you know after I run the tests.

File: src/FirstPayment/FirstPaymentBuilder.php (hunk 1 above)
Comment (chris-redbeed) [reply src/FirstPayment/FirstPaymentBuilder.php]: Moved it to `Laravel\Cashier\FirstPayment\Traits`

File: src/Traits/PaymentMethodString.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+trait PaymentMethodString
+{
+    /**
+     * Backwards compatible: split strings into array
+     *
+     * @param  string  $method
+     *
+     * @return string[]
+     */
+    private function castPaymentMethodString(string $method)
```
[the remark this answers is not in the document]
Comment (chris-redbeed) [reply src/Traits/PaymentMethodString.php]: I add also the `unique` filter

**Comment by sandervanhooft** [comment]
> Please see my comments in #203. The `Billable` trait needs to be updated too, in order to correctly work with the updated firstPaymentMethod return value in this PR, which now is a array instead of a string
> 
> I'm specifically pointing to [line 78](https://github.com/laravel/cashier-mollie/blob/develop/src/Billable.php#L78) in `Billable::class`

Not sure I get your point. Can you elaborate on this?

**Comment by sandervanhooft** [comment]
> > Please see my comments in #203. The `Billable` trait needs to be updated too, in order to correctly work with the updated firstPaymentMethod return value in this PR, which now is a array instead of a string
> > I'm specifically pointing to [line 78](https://github.com/laravel/cashier-mollie/blob/develop/src/Billable.php#L78) in `Billable::class`
> 
> Not sure I get your point. Can you elaborate on this?

All cleared up now in #203.

**Comment by sandervanhooft** [comment]
- Currently, the PlanContract states that `method` should return a string. People are relying on this, so we should keep this compatible.

- Also, the FirstPaymentBuilder::getMolliePayload is now returning a Collection instance instead of an array.

**Comment by sandervanhooft** [comment]
Let's treat this as a breaking change (BC) and adopt it in `v2`.

This also means we can simplify it further by changing everything into `array`, no need for backwards compatibility.

**Comment by salmanhijazi** [comment]
How to bypass this for now to allow multiple methods?

**Comment by sandervanhooft** [comment]
Merged into `v2-develop`, thanks @chris-redbeed !

**Pull request merged**